test/jdk/javax/swing/JTable/KeyBoardNavigation.java

```diff
@@ -224,13 +172,127 @@ public void setValueAt(Object aValue, int row, int column) {
     }
 
     public static void main(String[] args) throws Exception {
-        SwingUtilities.invokeAndWait(() -> {
-            try {
-                initTest();
-            } catch (Exception e) {
-                throw new RuntimeException(e);
-            }
-        });
-        passFailJFrame.awaitAndCheck();
+        String INSTRUCTIONS = """
+                Instructions to Test:
+                1. Refer the below keyboard navigation specs
+                 (referenced from bug report 4112270).
+                2. Check all combinations of navigational keys in all four modes
+                  shift and control verifying each change to the selection against
+                  the spec. If it does, press "pass", otherwise press "fail".
+
+                """;
+
+        INSTRUCTIONS += getOSSpecificInstructions();
+        PassFailJFrame.builder()
+                .instructions(INSTRUCTIONS)
+                .rows(11)
+                .columns(50)
+                .testUI(KeyBoardNavigation::initTest)
+                .build()
+                .awaitAndCheck();
+    }
+
+    public static String getOSSpecificInstructions() {
+        final String WINDOWS_SPECIFIC = """
+                Tab, Shift-Tab - Navigate In.
+                Return/Shift-Return - move focus one cell down/up.
+                Tab/Shift-Tab -  move focus one cell right/left.
+                Up/Down Arrow - deselect current selection; move focus one
+                                cell up/down
+                Left/Right Arrow - deselect current selection; move focus
+                                   one cell left/right
+                PageUp/PageDown - deselect current selection; scroll up/down
+                                  one JViewport view; first visible cell in
+                                  current column gets focus
+                Control-PageUp/PageDown - deselect current selection;
+                                          move focus and view to
+                                          first/last cell in current row
+                Home/End - deselect current selection; move focus and view to
+                           first/last cell in current row
+                Control-Home/End - deselect current selection;
+                                   scroll up/down one  JViewport view;
+                                   first/last visible row of the table
+                F2 - Allows editing in a cell containing information without
+                     overwriting the information
+                Esc -  Resets the cell content back to the state it was in
```

Comment:
```suggestion
                Esc -  Reset the cell content back to the state it was in.
```

Minor suggestion,  end with `.` for each statement else nothing.